### PR TITLE
Issue/16957 - Display scan error states

### LINKED ISSUE #16957: Jetpack App: display scan error states using a NoResultsVC
## Description

On the Jetpack Scan screen, the following error states should displayed in a NoResultsViewController:
- generic error
- multisites not available error 

## Notes

Currently, the error states are displayed using the same view component as non-error states. (Ref: https://github.com/wordpress-mobile/WordPress-iOS/pull/16817)

It would be better to display the error states using a NoResultsVC for a couple of reasons:
- Most screens use the NoResultsVC to display an error state 
- Using a NoResultsVC would match the error state UI on Android more closely

To reproduce the multisites not available error, please follow the **How to test** section in https://github.com/wordpress-mobile/WordPress-iOS/pull/16817

### Generic error
Current (iOS screenshot) | Expected (Android screenshot)
-- | --
 | 

### Multisite not supported error
Current (iOS screenshot) | Expected (Android screenshot)
-- | --
 |

### PR DESCRIPTION
**Pull request by nikola-milicevic**
Fixes [#16957](https://github.com/wordpress-mobile/WordPress-iOS/issues/16957)

### Description

This PR displays appropriate errors using `NoResultsViewController`.

### Result

- Multisite not supported error

Before | After
--- | ---
Screen Shot 2021-08-10 at 09 04 10 | Screen Shot 2021-08-09 at 16 56 39

### How to test

- Multisite not supported error

1. Go to the Jurassic Ninja special ops page
2. Select the **Jetpack Complete** plan
3. Launch a multisite on subdomains
4. Go to newly created site dashboard and click on **Set up Jetpack**
5. Inside Jetpack / Wordpress app select newly created site
6. Tap on **My Site** > **Scan**
6. Verify new design

### Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

### PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by nikola-milicevic** [comment]
What I'm not certain about is that if this PR fixes the `generic error` case. If someone can clarify that for me or take a look at the potential use cases that would be really helpful.

File: WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanCoordinator.swift [hunk 1]
```diff
@@ -7,6 +7,8 @@ protocol JetpackScanView {
     func showNoConnectionError()
     func showGenericError()
     func showScanStartError()
+    func showMultisiteNotSuportedError()
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanCoordinator.swift]:
```suggestion
    func showMultisiteNotSupportedError()
```

File: WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift [hunk 2]
```diff
@@ -79,28 +79,36 @@ class JetpackScanViewController: UIViewController, JetpackScanView {
...
+        let model = NoResultsViewController.Model(title: NoResultsText.scanStartError.title,
+                                                  subtitle: NoResultsText.scanStartError.subtitle,
+                                                  buttonText: NoResultsText.contactSupportButtonText)
         updateNoResults(model)
+    }
 
+    func showMultisiteNotSuportedError() {
+        guard let scanStatusViewModel = JetpackScanStatusViewModel(coordinator: coordinator) else {
+            return
+        }
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift]: What do you think of defining the string directly inside the NoResultsText struct?
Then, we can also delete the .multisitesNotSupported logic from `JetpackScanStatusViewModel`.

**Review by momo-ozawa — COMMENTED** [review]
First PR! 🎉 

The `multisites_not_supported` error is being displayed in a NoResultsVC 😄 

I'll leave a couple comments below.

**Comment by momo-ozawa** [comment]
> What I'm not certain about is that if this PR fixes the generic error case. If someone can clarify that for me or take a look at the potential use cases that would be really helpful.

So I think there are two cases where we want to show a generic error.
1. If getScanStatus fails (already being displayed in a NoResultsVC)
2. If getScanStatus succeeds and the most recent scan status has a `didFail` value of `true` (not yet being displayed in a NoResultsVC)

https://github.com/wordpress-mobile/WordPress-iOS/blob/81af355dac4021fafc746827dc30a5d31cb511ea/WordPress/Classes/ViewRelated/Jetpack/Jetpack%20Scan/View%20Models/JetpackScanStatusViewModel.swift#L191-L197

@emilylaguna Am I interpreting the second case correctly? Can you confirm that these are two distinct cases?

**Comment by momo-ozawa** [comment]
@nikola-milicevic Also, a couple pointers about PRs:
- Please make sure to include the Regression Notes and the PR submission checklist 🙇‍♀️ 
- For PRs, it's always good to have a How to test section even if it means copying and pasting from another PR 👍 
- It would be great if you could include a before / after screenshot, especially for UI related PRs 😄 
- To keep our bots happy, you'll need to assign a label and a milestone. Otherwise, you'll get [this error](https://github.com/wordpress-mobile/WordPress-iOS/pull/16998#issuecomment-895307839) 🤖 
    - Usually I just assign the same labels as the issues
    - Since 18.0 is in codefreeze, and this is an "enhancement" PR, you can assign this PR to the 18.1 milestone

**Comment by nikola-milicevic** [comment]
@momo-ozawa Thanks for the comments, I updated the code!

**Review by momo-ozawa — APPROVED** [review]
LGTM!!! 🥳

**Comment by momo-ozawa** [comment]
@nikola-milicevic 
Leaving a note here about our discussion on Slack: let's open a separate PR to tackle the generic errors if necessary. 👍

**Pull request merged**